crucible-llvm/src/Lang/Crucible/LLVM/Translation/BlockInfo.hs

```diff
@@ -0,0 +1,280 @@
+-----------------------------------------------------------------------
+-- |
+-- Module           : Lang.Crucible.LLVM.Translation.BlockInfo
+-- Description      : Pre-translation analysis results
+-- Copyright        : (c) Galois, Inc 2018-2021
+-- License          : BSD3
+-- Maintainer       : Rob Dockins <rdockins@galois.com>
+-- Stability        : provisional
+-----------------------------------------------------------------------
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE ImplicitParams        #-}
+{-# LANGUAGE PatternSynonyms       #-}
+{-# LANGUAGE RankNTypes            #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeOperators         #-}
+
+module Lang.Crucible.LLVM.Translation.BlockInfo
+  ( LLVMBlockInfoMap
+  , LLVMBlockInfo(..)
+  , buildBlockInfoMap
+  , useTypedVal
+  ) where
+
+import Data.Foldable (toList)
+import Data.List (foldl')
+import Data.Maybe
+import Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+import Data.Sequence (Seq)
+import qualified Data.Sequence as Seq
+import Data.Set (Set)
+import qualified Data.Set as Set
+
+import qualified Text.LLVM.AST as L
+
+import           Lang.Crucible.CFG.Generator
+import           Lang.Crucible.Panic ( panic )
+
+type LLVMBlockInfoMap s = Map L.BlockLabel (LLVMBlockInfo s)
+
+-- | Information about an LLVM basic block computed before be we begin the
+--   translation proper.
+data LLVMBlockInfo s
+  = LLVMBlockInfo
+    {
+      -- | The crucible block label corresponding to this LLVM block
+      block_label :: Label s
+
+      -- | The computed "use" set for this block.  This is the set
+      -- of identifiers that must be assigned prior to jumping to this
+      -- block. They are either used directly in this block or used
+      -- by a successor of this block.  Note! "metadata" nodes
+      -- do not contribute to the use set.  Note! values referenced
```

Comment:
It would be good to explain somewhere _why_ metadata notes do not contribute to the use set, citing the relevant LLVM ticket.